module/zfs/zvol.c

```diff
@@ -1589,7 +1589,11 @@ zvol_probe(dev_t dev, int *part, void *arg)
 	struct kobject *kobj;
 
 	zv = zvol_find_by_dev(dev);
+#ifndef HAVE_GET_DISK_AND_MODULE
 	kobj = zv ? get_disk(zv->zv_disk) : NULL;
+#else
+	kobj = zv ? get_disk_and_module(zv->zv_disk) : NULL;
+#endif
```

Comment:
nit: let's move this in to `include/linux/blkdev_compat.h`, maybe something alone the lines of `get_disk_ro`.

```c
#ifndef HAVE_GET_DISK_AND_MODULE
static inline struct kobject *
get_disk_and_module(struct gendisk *disk)
{
        return (get_disk(disk));
}
#endif
```